src/Shared/Components/Plugin/utils.tsx

```diff
@@ -0,0 +1,124 @@
+import { ParentPluginDTO, PluginCreationType, PluginDataStoreType } from './types'
+import { commonSelectStyles } from '../ReactSelect'
+import { stringComparatorBySortOrder } from '../../Helpers'
+import { DEFAULT_PLUGIN_CREATED_BY } from './constants'
+
+const parseMinimalPluginVersionsDTO = (
+    pluginVersionData: ParentPluginDTO['pluginVersions']['minimalPluginVersionData'],
+): PluginDataStoreType['parentPluginStore'][number]['pluginVersions'] => {
+    if (!pluginVersionData?.length) {
+        return []
+    }
+
+    return pluginVersionData.map(({ id, description, name, pluginVersion, isLatest }) => ({
+        id,
+        description: description || '',
+        name: name || '',
+        pluginVersion: pluginVersion || '',
+        isLatest: isLatest || false,
+    }))
+}
+
+export const parsePluginDetailsDTOIntoPluginStore = (pluginData: ParentPluginDTO[]): PluginDataStoreType => {
+    if (!pluginData?.length) {
+        return {
+            parentPluginStore: {},
+            pluginVersionStore: {},
+        }
+    }
+
+    const parentPluginStore: PluginDataStoreType['parentPluginStore'] = {}
+    const pluginVersionStore: PluginDataStoreType['pluginVersionStore'] = {}
+
+    pluginData.forEach((plugin) => {
+        const pluginVersions = parseMinimalPluginVersionsDTO(plugin.pluginVersions.minimalPluginVersionData)
+        const latestPluginVersionIndex = pluginVersions.findIndex((pluginVersion) => pluginVersion.isLatest)
+
+        // Adding this check to ensure better debugging in case backend panics
+        if (latestPluginVersionIndex === -1) {
+            throw new Error('Latest plugin version not found')
```

Comment:
Log to sentry as well

Reply:
Where it is being used, it wrapped in try catch block with error thrown to showError and that would automatically captureException to sentry